geode-core/src/main/java/org/apache/geode/internal/datasource/GemFireTransactionDataSource.java

```diff
@@ -247,4 +255,12 @@ public void close() {
     super.close();
     provider.clearUp();
   }
+
+  /**
+   * Used by unit tests
+   */
+  @VisibleForTesting
+  public void setTransManager(TransactionManager txmanager) {
```

Comment:
If setTransManager is invoked from a unit test then most like that unit test is in the same package. If so then remove public so it's package-private (default scope).

Reply:
Updated